packages/titus-backend/lib/plugins/auth/authzcheck.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+const { Forbidden } = require('http-errors')
+const fp = require('fastify-plugin')
+
+async function authzChecks(server, options) {
+  server.route({
+    method: 'GET',
+    url: '/admin-authzcheck',
+    schema: {
+      tags: ['authz'],
+      security: [
+        {
+          apiKey: []
+        }
+      ]
+    },
+    onRequest: [
+      server.authenticate,
+      async (request, res) => {
+        const { auth } = options
+        let isAdmin = false
+
+        // this is a sample to check the signed-in user's permission to access this resource via casbin policies based on configuration in the OAuth provider
+        // a real-world example would have more context around the user account and roles/scopes/permissions and policies
+
+        if (auth.provider === 'auth0') {
+          // depending on how this is managed - you could get associated user data from auth0 user management API like https://auth0.com/docs/api/management/v2#!/Users/get_user_roles
+          // or set this up to be included in the token via Auth0 rules https://auth0.com/docs/rules
+          isAdmin = true
```

Comment:
always true for now - involves some set up in Auth0, or another way to associate the user with `admin`